core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -109,6 +111,9 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(classOf[SerializableJobConf], new KryoJavaSerializer())
     kryo.register(classOf[HttpBroadcast[_]], new KryoJavaSerializer())
     kryo.register(classOf[PythonBroadcast], new KryoJavaSerializer())
+    kryo.register(classOf[TaskMetrics], new KryoJavaSerializer())
+    kryo.register(classOf[DirectTaskResult[_]], new KryoJavaSerializer())
+    kryo.register(classOf[IndirectTaskResult[_]], new KryoJavaSerializer())
```

Comment:
Moreover, `_accumulatorUpdates` is much more complicated. Its value could be any type, e.g., `SQLMetricValue`. If someone adds a new `SQLMetricValue` type, it's hard for reviewers to recall it will be stored in `TaskMetrics`.